python/ray/autoscaler/commands.py

```diff
@@ -856,18 +856,15 @@ def exec_cluster(config_file: str,
             runtime_hash="",
             file_mounts_contents_hash="",
             is_head_node=True,
-            docker_config=config.get("docker"))
-
-        is_docker = isinstance(updater.cmd_runner, DockerCommandRunner)
+            docker_config=docker_config)
 
+        is_docker = docker_config and docker_config["container_name"] != ""
         if cmd and stop:
             cmd += "; ".join([
                 "ray stop",
                 "ray teardown ~/ray_bootstrap_config.yaml --yes --workers-only"
             ])
-            if is_docker and run_env == "docker":
-                updater.cmd_runner.shutdown_after_next_cmd()
```

Comment:
I don't find this function in the DockerCommandRunner at all, how did this pass the tests? @ijrsvt @ericl 
The function in DockerCommandRunner is `_shutdown_after_next_cmd`, which is never called. I removed the function but we probably need another patch for this (it is already not working and we want to remove this function).

Reply:
Ah, this should actually be `_shutdown_afer_next_cmd()`. This is for cases where a user wants the cluster to shutdown after a command is finished executing.

Reply:
We can't call `._shutdown_after_next_cmd()` for two reasons:
1) It breaks the python class convention of _XXX functions should not be called from outside the class.
2) It violates the CommandRunnerInterface abstractions.
@ijrsvt , can you please think of ways to get rid of this other than adding this to the CommandRunnerInterface?

Reply:
Actually, I am okay with raising an error if the CommandRunner is `DockerCommandRunner` or `KubernetesCommandRunner` when `stop` is passed. This warning can read:
```
Running stop when using a Containerized environment will shutdown all worker nodes, but not the head node. Please manually run ray stop after the workload is finished running.
```

I think we can just create an issue to fix this in the future, because we need to think about how to handle `stop` when `tmux`/`screen` is passed in. The idea of just adding `sudo shutdown -h` will not work in a containerized environment.